### DISCUSSION TITLE
Do node numbers have to be contiguous?

### DISCUSSION THREAD

**Discussion by ufoozhenghao** [pyg-team/pytorch_geometric#9255] [Q&A]
I am trying to create my own dataset and it's kind of like [METRLADatasetLoader](https://pytorch-geometric-temporal.readthedocs.io/en/latest/modules/dataset.html?highlight=METRLADatasetLoader#torch_geometric_temporal.dataset.metr_la.METRLADatasetLoader) 

My model is based on PyTorch Geometric Temporal Model. I want to know if node numbers have to be contiguous（0,1,2,3,4...）?

And in validation set my data will get from different sensors. If some of them were broken or can't get data, which symbol should I use to instead?

**Comment by rusty1s** [comment]
Yes, node numbers need to be between `[0, num_nodes)`. If you have missing nodes in validation, you can either discard them, or simply give them a zero-feature vector.
